streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -234,10 +256,10 @@ private ClientResponse sendRequest(final ClientRequest clientRequest) {
     /**
      * Fetch the metadata for all topics
      */
-    public Collection<MetadataResponse.TopicMetadata> fetchTopicsMetadata() {
+    public MetadataResponse fetchMetadata() {
 
         final ClientRequest clientRequest = kafkaClient.newClientRequest(
-            getBrokerId(),
+            getAnyReadyBrokerId(),
```

Comment:
This is the code before creating topics. This code just gets the metadata first, which then allows us to figure out which one is the controller.